File: 02_Object_Oriented_Programming/05_Polymorphism_and_It's_types/01_Compile_Time_Polymorphism/01_Example_1/javascript/vehicle.js
class Vehicle {
    // JavaScript simulates method overloading using default/optional parameters
    start(vehicleType, speed = null) {
        if (speed !== null) {
            console.log(`Starting a ${vehicleType} with speed: ${speed} km/h`);
        } else {
            console.log(`Starting a ${vehicleType}`);
        }
    }
}

module.exports = Vehicle;

File: 02_Object_Oriented_Programming/05_Polymorphism_and_It's_types/01_Compile_Time_Polymorphism/01_Example_1/python/vehicle.py
class Vehicle:
    # Python simulates method overloading using default parameters
    def start(self, vehicle_type, speed=None):
        if speed is not None:
            print(f"Starting a {vehicle_type} with speed: {speed} km/h")
        else:
            print(f"Starting a {vehicle_type}")

